expertsystem/ui/__init__.py

```diff
@@ -72,6 +74,7 @@ def __init__(  # pylint: disable=too-many-arguments
                 f"Formalism type {formalism_type} not implemented."
                 f" Use {allowed_formalism_types} instead."
             )
+        self.__formalism_type = formalism_type
```

Comment:
Actually a lot of the other attributes should also be made into private variables or properties etc

Reply:
Yep... working on that

Reply:
but better to do that in a separate PR

Reply:
ok